### ISSUE TITLE
False negative with NP_METHOD_PARAMETER_TIGHTENS_ANNOTATION using @NonNullByDefaul

### ISSUE THREAD

**Issue by ghernadi** [spotbugs/spotbugs#3335]
Basically the same as #3324 but for parameters:

`package-info.java`:
```java
@NonNullByDefault
package ghIssues.issue3335;

import org.eclipse.jdt.annotation.NonNullByDefault;
```

```java
package ghIssues.issue3335;

import javax.annotation.Nullable;

public interface DummyInterface {
    void setNullable1(@Nullable Object obj);
    void setNullable2(@Nullable Object obj);
}
```

```java
package ghIssues.issue3335;

import javax.annotation.Nonnull;

public class DummyImpl implements DummyInterface {
    @Override
    public void setNullable1(Object objRef)  {}

    @Override
    public void setNullable2(@Nonnull Object objRef) {}
}
```

I would have expected both methods in `DummyImpl` to report `NP_METHOD_PARAMETER_TIGHTENS_ANNOTATION`.
